go/merge/three_way.go

```diff
@@ -200,7 +190,81 @@ func threeWayStructMerge(a, b, parent types.Struct, vwr types.ValueReadWriter, p
 		}
 		panic(fmt.Errorf("Bad key type in diff: %s", change.V.Type().Describe()))
 	}
-	return threeWayOrderedSequenceMerge(parent, aDiff, bDiff, makeGetFunc(a), makeGetFunc(b), makeGetFunc(parent), apply, vwr, progress)
+	return m.threeWayOrderedSequenceMerge(structCandidate{a}, structCandidate{b}, structCandidate{parent}, apply, path)
+}
+
+type candidate interface {
```

Comment:
I was explaining while you were reviewing!

The 'candidate' stuff was because the alternative was passing something like 7 closures to `threeWayOrderedSequenceMerge()`: one getFunc for each of a, b, and parent; one diffFunc for each of a and b; and one path-concatenation function for each of a and b.

That seemed absurd, so instead I made a little interface that Maps, Sets and Structs could all be adapted to and made the function take those instead. mapCandidate, setCandidate, and structCandidate are the adaptor implementations.